src/engine/Cache.class.js

```diff
@@ -421,36 +428,78 @@ class Cache {
       const archivedFilename = path.basename(filePath)
       const archivePath = path.join(this.archiveFolder, archivedFilename)
 
-      switch (this.archiveMode) {
-        case 'delete':
-          // Delete original file
-          fs.unlink(filePath, (unlinkError) => {
-            if (unlinkError) {
-              this.logger.error(unlinkError)
-            } else {
-              this.logger.info(`File ${filePath} deleted`)
-            }
-          })
-          break
-        case 'archive':
-          // Create archive folder if it doesn't exist
-          if (!fs.existsSync(this.archiveFolder)) {
-            fs.mkdirSync(this.archiveFolder, { recursive: true })
+      if (this.archiveMode) {
+        // Create archive folder if it doesn't exist
+        if (!fs.existsSync(this.archiveFolder)) {
+          fs.mkdirSync(this.archiveFolder, { recursive: true })
+        }
+
+        // Move original file into the archive folder
+        fs.rename(filePath, archivePath, (renameError) => {
+          if (renameError) {
+            this.logger.error(renameError)
+          } else {
+            this.logger.info(`File ${filePath} moved to ${archivePath}`)
           }
+        })
+      } else {
+        // Delete original file
+        fs.unlink(filePath, (unlinkError) => {
+          if (unlinkError) {
+            this.logger.error(unlinkError)
+          } else {
+            this.logger.info(`File ${filePath} deleted`)
+          }
+        })
+      }
+    }
+  }
 
-          // Move original file into the archive folder
-          fs.rename(filePath, archivePath, (renameError) => {
-            if (renameError) {
-              this.logger.error(renameError)
-            } else {
-              this.logger.info(`File ${filePath} moved to ${archivePath}`)
-            }
-          })
-          break
-        default:
-          this.logger.error(`unknown Archive Mode: ${this.archiveMode}`)
+  /**
+   * Delete file in archiveFolder
+   * @param {string} filePath - The file
+   * @return {void}
+   */
+  async refreshArchiveFolder() {
+    this.logger.silly('refreshArchiveFolder()')
+    // if a process already occurs, it clears it
+    if (this.archiveTimeout) {
+      clearTimeout(this.archiveTimeout)
+    }
+
+    const files = fs.readdirSync(this.archiveFolder)
+    const timestamp = new Date().getTime()
+    if (files.length > 0) {
+      // Disable ESLint check because we need for..of loop to support async calls
+      // eslint-disable-next-line no-restricted-syntax
+      for (const file of files) {
+        // Disable ESLint check because we want to handle files one by one
+        // eslint-disable-next-line no-await-in-loop
+        const stats = fs.statSync(path.join(this.archiveFolder, file))
+
+        if (stats.mtimeMs + this.retentionDuration < timestamp) {
```

Comment:
this.retentionDuration is in seconds (you multiply it by 3600 in the constructor).  stats.mtimeMs give you milliseconds. You must change multiply by 3600000 in the constructor instead